wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java

```diff
@@ -382,8 +382,15 @@ private static <T extends RobotBase> void runRobot(Supplier<T> robotSupplier) {
       }
       String robotName = "Unknown";
       StackTraceElement[] elements = throwable.getStackTrace();
-      if (elements.length > 0) {
-        robotName = elements[0].getClassName();
+      for (var element : elements) {
+        try {
+          if (RobotBase.class.isAssignableFrom(Class.forName(element.getClassName()))) {
+            robotName = element.getClassName();
+            break;
+          }
+        } catch (ClassNotFoundException e) {
+          // The class is in the stack trace, it better exist
```

Comment:
Even though we don't expect it to happen, it might be nice to have *some* indication of what happened instead of just silently using `Unknown`.